dds/DCPS/Serializer.h

```diff
@@ -669,25 +669,30 @@ class OpenDDS_Dcps_Export Serializer {
 
   void set_construction_status(ConstructionStatus cs);
 
+  struct OpenDDS_Dcps_Export SavePoint {
+    explicit SavePoint(Serializer& ser);
+
+    void restore(Serializer& ser) const;
+
+    const size_t pos_;
+    ACE_Message_Block* const current_;
+    char* const rd_ptr_;
+    char* const wr_ptr_;
+  };
```

Comment:
While this does preserve the functionality that was in `peek()` before this PR, I don't think this is sufficient to save and restore the Serializer's state.  The alignment support is not considered here (like `align_rshift_`).

I think we should use a different design for generalized `peek`.  Construct a new Serializer object using a duplicated message block.  These are both lightweight operations -- just updating reference counts and other scalars.

Before this PR, `peek()` is only used for the XCDR2 `LC > 4` case in `Serializer::read_parameter_id`.  In that specific case I don't know if we need a new Serializer object or if we should just have a peek_uint32 as a fundamental operation on Serializer.

Reply:
Gotcha. I saw that this was only used in that one place. Do you want me to nuke the template for now and just make the a uint32 version? Or leave the template and add a comment / make a Jira ticket? Or try to fix the template and also add the uint32 version?

Reply:
I mean, could the SavePoint just save/restore `align_rshift_` as well and that fixes the problem?

Reply:
Looking at Serializer, it might also want to grab `align_wshift_`, `good_bit_`, `encoding_`, `swap_bytes_` ... anything else?

Reply:
and `construction_status_` ... I think that's it?

Reply:
`duplicate()` itself has an allocation so if that needs to be avoided, the `message_block_allocator_` would be needed.

I think the align_rshift_ approach is fine.  There is a unit test for Serializer which should be extended for this, including the corner cases of advancing the message block chain to a block that has a different alignment.